### PR TITLE
Executable padvolume

### PR DESCRIPTION
**Pull request by archanasri**

### CODE REVIEW

**Comment by akenmorris** [comment]
The automated python test (e.g. ellipsoid) is failing with:

Traceback (most recent call last):
3:   File "ellipsoidMain.py", line 45, in <module>
3:     Run_Ellipsoid_Pipeline(args)
3:   File "/home/runner/work/ShapeWorks/ShapeWorks/Examples/Python/RunEllipsoid.py", line 105, in Run_Ellipsoid_Pipeline
3:     resampledFiles = applyIsotropicResampling(parentDir + "resampled", fileList)
3:   File "/home/runner/work/ShapeWorks/ShapeWorks/Examples/Python/GroomUtils.py", line 46, in applyIsotropicResampling
3:     if binary:
3: NameError: name 'binary' is not defined

There seems to be a mismatch between "binary" and "isBinary"

File: Applications/shapeworks/Commands.cpp [hunk 1]
```diff
@@ -263,6 +287,34 @@ int SmoothMesh::execute(const optparse::Values &options, SharedCommandData &shar
   return sharedData.mesh.smooth(/*maxRMSErr, numIter*/);
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// Coverage
+///////////////////////////////////////////////////////////////////////////////
+void Coverage::buildParser()
+{
+  const std::string prog = "coverage";
+  const std::string desc = "coverage between two meshes";
```
Comment (cchriste) [inline Applications/shapeworks/Commands.cpp]: This is not a sufficient description. What does "coverage between two meshes" mean? Maybe _compute_ coverage between two meshes?

Comment (cchriste) [reply Applications/shapeworks/Commands.cpp]: What is the result? Another mesh?

Comment (archanasri) [reply Applications/shapeworks/Commands.cpp]: I did not create the coverage command

File: Applications/shapeworks/Commands.cpp [hunk 2]
```diff
@@ -263,6 +287,34 @@ int SmoothMesh::execute(const optparse::Values &options, SharedCommandData &shar
...
+  parser.prog(prog).description(desc);
+  parser.add_option("--second_mesh").action("store").type("string").set_default("").help("Second mesh to apply coverage.");
```
Comment (cchriste) [inline Applications/shapeworks/Commands.cpp]: What is the "second_mesh"? Is it a target to which "coverage" is applied? Is it therefore the result of the command? There just isn't enough information here for the command to be understood. I'm creating an issue about this (add detailed documentation shown only for <subcommand> --help), but for now maybe better to add a bit more description.

File: Applications/shapeworks/notes.txt [hunk 3]
```diff
@@ -63,7 +65,12 @@ Image commands:
 [] *ExtractGivenLabelImage
 [] *FastMarching
 [] *FindLargestBoundingBox
-[] *PadVolumeWithConstant
+[] *PadVolumeWithConstant -> shapeworks padvolume
```
Comment (cchriste) [inline Applications/shapeworks/notes.txt]: If it's done let's put an 'x' in here. 👍

File: Applications/shapeworks/Commands.cpp (hunk 1 above)
Comment (akenmorris) [reply Applications/shapeworks/Commands.cpp]: I will ask Praful & Steve how this should be described.

File: Examples/Scripts/Tools/PadVolumes.sh [hunk 4]
```diff
@@ -124,9 +124,9 @@ do
     EchoWithColor  "imgfilename_padded $imgfilename_padded" "yellow"
     EchoWithColor "-------------------------------------------------------------------------------------------------" "yellow"
     
-    PadVolumeWithConstant --inFilename $segfilename --outFilename $segfilename_padded --paddingSize $padding_size --paddingValue $padVal
+    shapeworks readimage --name $segfilename padimage writeimage --name $segfilename_padded
```
Comment (cchriste) [inline Examples/Scripts/Tools/PadVolumes.sh]: $padVal is no longer used. Please add it back to the commands here.

File: Libs/Image/Image.cpp [hunk 5]
```diff
@@ -286,10 +286,29 @@ bool Image::resample(float isoSpacing, PixelType defaultvalue, Dims outputSize)
 ///////////////////////////////////////////////////////////////////////////////
 bool Image::compare_equal(const Image &other)
 {
+  // we use the region of interest filter here with the full region because our
```
Comment (cchriste) [inline Libs/Image/Image.cpp]: Is there another comparison function? How do we now test functions like `recenter` which do not modify the image, only its origin?

Comment (akenmorris) [reply Libs/Image/Image.cpp]: I believe it will still work for a function such as `recenter`.

File: Libs/Mesh/Mesh.cpp [hunk 6]
```diff
@@ -1,66 +1,80 @@
 #include "Mesh.h"
+#include <PreviewMeshQC/FEAreaCoverage.h>
+#include <PreviewMeshQC/FEVTKImport.h>
+#include <PreviewMeshQC/FEVTKExport.h>
+
+#include <vtkPolyDataReader.h>
+#include <vtkPolyDataWriter.h>
 
 //#include <vtkVersion.h>
```
Comment (cchriste) [inline Libs/Mesh/Mesh.cpp]: This is great that the Mesh class implementation is starting. Please clean it up: no commented code like was in there, and either implement or completely drop the functions that were simply placeholders before. Thank you!

File: Libs/Mesh/Mesh.h [hunk 7]
```diff
@@ -21,11 +20,14 @@ class Mesh
   /// \param filename
   bool write(const std::string &outFilename);
 
+  /// coverage
```
Comment (cchriste) [inline Libs/Mesh/Mesh.h]: Please emulate Image.h/cpp and move the function comments/descriptions to the .cpp file.

File: Testing/ImageTests/ImageTests.cpp [hunk 8]
```diff
@@ -19,6 +19,43 @@ TEST(ImageTests, resample_test) {
...
+  Image ground_truth(test_location + "1x2x2.nrrd");
+
+  ASSERT_TRUE(image.compare_equal(ground_truth));
+}
+
+TEST(ImageTests, binarize_test) {
+
+std::string test_location = std::string(TEST_DATA_DIR) + std::string("/binarize/");
+
+Image image(test_location + "1x2x2.nrrd");
```
Comment (cchriste) [inline Testing/ImageTests/ImageTests.cpp]: As we were learning the hard way before, let's limit these tests to one per operation. Or we can figure out some way to indicate to the developer whose test fails which part of the test that failed (maybe googletests has some way to indicate "subtests" within a test).

File: Applications/shapeworks/notes.txt (hunk 3 above)
Comment (archanasri) [reply Applications/shapeworks/notes.txt]: Done

**Pull request merged**